internal/config/config.go

```diff
@@ -62,6 +62,7 @@ type authConfig struct {
 	OpenShiftApiUrl       string `json:"openShiftApiUrl,omitempty"`
 	OIDCAuthority         string `json:"oidcAuthority,omitempty"`
 	InternalOIDCAuthority string `json:"internalOidcAuthority,omitempty"`
+	AAPGatewayApiUrl      string `json:"aapGatewayApiUrl,omitempty"`
```

Comment:
This authConfig struct is getting a bit messy, in particular that we add more and more fields for each AuthN/Z provider and do not check for incompatible or invalid config. Could we at a minimum add input validation for the auth in https://github.com/flightctl/flightctl/blob/main/internal/config/config.go#L173-L175, please? (I'll open a Jira for non-auth partS.)

Also wondering whether we can sub-structure a bit more. For example, ATM the OpenShiftApiUrl and AAPGatwayApiUrl assume that's the URL for everything: proxy, authN, and authZ. But down these may be different (e.g. using OpenShift for Auth, but SpiceDB instead of K8s RBAC), so wondering what are common vs provider-specific things that would belong into a sub-struct.

Reply:
how about
```yaml
auth:
  // common props
  caCert: string
  insecureSkipTlsVerify: bool
  // provider specific props
  k8s:
    apiUrl: string
    externalOpenShiftApiUrl: string
    //additional config ie spiceDB
    spiceDB:
      foo: string
  odic:
    oidcAuthorityUrl: string
    externalOidcAuthorityUrl: string
  aap:
    gatewayUrl: string
    ...
```